### PR TITLE
Aj/86bz12uff/login forn

### PR DESCRIPTION
**Pull request by Ahmedjaziri3**
## What does this do?

This pull request enhances the supplier login interface in the Archimatch app:

- **Added Assets:** Introduced a new PNG file ("supplier login.png") for visual representation.
- **Updated Components:** Modified `LoginForm` component in `login-form-supplier.tsx` to incorporate controlled inputs for email validation using `react-hook-form`.
- **Improved User Experience:** Implemented responsive design using `KeyboardAvoidingView` to ensure optimal display on both iOS and Android platforms.

## Why did you do this?

These changes aim to streamline the supplier login process, improving usability and visual appeal across different devices accessing Archimatch.

## Who/what does this impact?

- **Developers:** Team members working on related components should review the changes, particularly those handling form submissions and UI updates.
- **Users:** Suppliers using the Archimatch app will experience a more intuitive and responsive login interface.

## How did you test this?

- Verified functionality of `react-hook-form` integration with controlled inputs for email validation.
- Tested responsiveness and layout consistency using `KeyboardAvoidingView` on iOS and Android devices.
- Confirmed correct rendering of the new PNG file ("supplier login.png") across various screen resolutions.

These steps ensure the reliability and usability of the updated supplier login feature in Archimatch.
## Some photos showcasing the changes :

325e1c16-b851-479c-858b-b53f620d3736

4befeac9-f1cd-44bc-b344-8d0dbdac060d

### CODE REVIEW

File: src/modules/login/login-form-supplier.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <>
+      <KeyboardAvoidingView
+        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+        className={`flex-1  items-center justify-start  bg-secondary`}
+        style={{ marginTop: '20%' }}
+      >
+        <Text
+          className={`text-lato-regular my-2 flex-row items-center justify-between text-xl font-bold`}
+        >
+          {'Bienvenue sur Archimatch'}
```
Comment (MohamedMegdich1) [inline src/modules/login/login-form-supplier.tsx:33]: add the translation for this line

File: src/modules/login/login-form-supplier.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        </Text>
+        <Text
+          className={`text-lato-regular flex-row items-center justify-between text-xl font-bold`}
+        >
+          {'Kit Partenariat PRO'}
```
Comment (MohamedMegdich1) [inline src/modules/login/login-form-supplier.tsx:38]: the same

File: src/app/(supplier)/(public)/login.tsx [hunk 3]
```diff
@@ -1,21 +1,14 @@
-import { useRouter } from 'expo-router';
 import React from 'react';
 
-import { useAuth } from '@/core';
 import { useSoftKeyboardEffect } from '@/core/keyboard';
-import type { LoginFormProps } from '@/modules/login/login-form';
-import { LoginForm } from '@/modules/login/login-form';
+import type { LoginFormProps } from '@/modules/login/login-form-supplier';
```
Comment (MohamedMegdich1) [inline src/app/(supplier)/(public)/login.tsx:4]: The type should be in the folder types

File: src/app/(supplier)/(public)/_layout.tsx [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
 
+import { BackButton } from '@/modules/shared';
+
 export default function SupplierPublicLayout() {
   return (
     <Stack initialRouteName="login">
-      <Stack.Screen name="login" options={{ headerShown: false }} />
+      <Stack.Screen
+        name="login"
+        options={{
```
Comment (MohamedMegdich1) [inline src/app/(supplier)/(public)/_layout.tsx:11]: It would be best if you changed the option from the branch of Oussama

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/modules/login/login-form-supplier.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+          className={`text-lato-regular flex-row items-center justify-between text-xl font-bold`}
+        >
+          {translate('loginSupplier.Kit_Partenariat_PRO')}
+        </Text>
+        <Container style="flex w-[100%] h-[28%] items-center mb-0 justify-between gap-3 bg-white  pb-2">
+          <ImageContainer className="flex  w-full items-center justify-center">
+            <Image
+              className="h-full w-full overflow-hidden rounded-t-xl "
+              contentFit="cover"
+              source={require('@/assets/Supplier login.png')}
```
Comment (MohamedMegdich1) [inline src/modules/login/login-form-supplier.tsx:45]: why the image name with space

File: src/translations/fr/loginSupplier.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "E-mail": "E-mail",
+  "Supporting Text": "Texte de support",
```
Comment (MohamedMegdich1) [inline src/translations/fr/loginSupplier.json:3]: Remember to follow the convention for naming translations.

File: src/translations/en/loginSupplier.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "E-mail": "Email",
+  "Supporting Text": "Supporting Text",
+  "Se connecter": "Login",
```
Comment (MohamedMegdich1) [inline src/translations/en/loginSupplier.json:4]: Remember to follow the convention for naming translations.

File: src/translations/ar/loginSupplier.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "E-mail": "البريد الإلكتروني",
+  "Supporting Text": "نص داعم",
+  "Se connecter": "تسجيل الدخول",
```
Comment (MohamedMegdich1) [inline src/translations/ar/loginSupplier.json:4]: Remember to follow the convention for naming translations.

File: src/modules/login/login-form.test.tsx [hunk 9]
```diff
@@ -56,7 +56,7 @@ describe('LoginForm Form ', () => {
     // undefined because we don't use second argument of the  SubmitHandler
     expect(onSubmitMock).toHaveBeenCalledWith(
       {
-        email: 'example@gmail.com',
+        email: 'example@gmail.com', // Corrected expected value to match input
```
Comment (MohamedMegdich1) [inline src/modules/login/login-form.test.tsx:59]: Extract it into a constant for better readability

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/modules/login/login-form-supplier.tsx (hunk 5 above)
Comment (Ahmedjaziri3) [reply src/modules/login/login-form-supplier.tsx:45]: Waiting for the .gif version

**Pull request closed without merging**